modules/ixBidAdapter.js

```diff
@@ -238,16 +270,59 @@ function isValidSize(size) {
  * @return {boolean}          True if the size object is an element of the size array, and false
  *                            otherwise.
  */
-function includesSize(sizeArray, size) {
+function includesSize(sizeArray = [], size) {
   if (isValidSize(sizeArray)) {
     return sizeArray[0] === size[0] && sizeArray[1] === size[1];
   }
-
   for (let i = 0; i < sizeArray.length; i++) {
     if (sizeArray[i][0] === size[0] && sizeArray[i][1] === size[1]) {
       return true;
     }
   }
+  return false;
+}
+
+/**
+ * Checking if all required video params are present
+ * @param {*} bid               Bid Object
+ * @param {*} mediaTypeVideoRef Ad unit level mediaTypes object
+ * @param {*} paramsVideoRef    IX bidder params level video object
+ * @returns bool                Are the required video params available
+ */
+function checkVideoParams(bid, mediaTypeVideoRef, paramsVideoRef) {
+  let isParamsLevelValid = true;
+  const requiredIXParams = ['mimes', 'minduration', 'maxduration', 'protocols'];
+
+  if (!mediaTypeVideoRef) {
+    utils.logWarn('IX Bid Adapter: mediaTypes.video is the preferred location for video params in ad unit');
+  }
+
+  for (let property of requiredIXParams) {
```

Comment:
about protocols :
Theres a deprecated `protocol` in openrtb and the current `protocols` .
IX adapter supports both.
So we need to check if either of the 2 are present in either level.

Also ill be wary of using `!propInMediaType && !propInVideoRef`, they could have a legit zero value ...

Reply:
Ah, that's a subtle difference that seems easy to miss...
Maybe we should move protocols out of "required parameters" since it isn't actually required and we should handle the protocol/protocols check more explicitly after this more trivial for loop? It isn't that pretty, maybe you can find a cleaner way to write it. Seems like a clearer logical separation of required and also makes the protocol vs protocols more obvious as they are next to each other.

```
for (let property of requiredIXParams) {
  const propInMediaType = mediaTypeVideoRef && mediaTypeVideoRef[property]
  const propInVideoRef = paramsVideoRef && paramsVideoRef[property]

  if (!propInMediaType && !propInVideoRef) {
    utils.logError('IX Bid Adapter: ' + property + ' is not included in either the adunit or params level');
    return false;
  }
}

const protocolMediaType = mediaTypeVideoRef && mediaTypeVideoRef.protocol
const protocolsMediaType = mediaTypeVideoRef && mediaTypeVideoRef.protocols
const protocolVideoRef = paramsVideoRef && paramsVideoRef.protocol
const protocolsVideoRef = paramsVideoRef && paramsVideoRef.protocols

return protocolMediaType || protocolsMediaType || protocolVideoRef || protocolsVideoRef
```

Reply:
This would fail in the case that any of the values is Zero .
on the line `if (!propInMediaType && !propInVideoRef) {` 
if  `minduration`  is Zero .

Id like to revisit this in a new PR, after considering deprecating the `protocol` in favor of just `protocols`

Reply:
Happy to let you revisit this. Just wanted to flag. Ack on the zero coercion issue, could still use something like hasOwnPropoerty. Point was just making it more readable.